src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -285,6 +285,8 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun checkBinary(holder: AnnotationHolder, o: RsBinaryExpr) {
         if (o.isComparisonBinaryExpr() && (o.left.isComparisonBinaryExpr() || o.right.isComparisonBinaryExpr())) {
             holder.createErrorAnnotation(o, "Chained comparison operator require parentheses")
+        } else if(o.isAssignBinaryExpr() && !o.left.isMutable()) {
+            holder.createErrorAnnotation(o, "Reassigning an immutable variable [E0384]")
```

Comment:
The exact `rustc` message is "re-assignment of immutable variable `foo`", so I'd suggest to use "Re-assignment of immutable variable `foo` [E0384]". A simpler form would be without the variable name, though it sometimes useful to have it right in the error message.